crates/client/src/storage/fs.rs

```diff
@@ -47,13 +47,16 @@ impl FileSystemRegistryStorage {
     /// If the lock cannot be acquired, `Ok(None)` is returned.
     pub fn try_lock(base_dir: impl Into<PathBuf>) -> Result<Option<Self>> {
         let base_dir = base_dir.into();
-        let mut registries_dir = base_dir.clone();
+        let registries_dir = &mut base_dir
+            .parent()
+            .context("base_dir cannot be empty")?
+            .to_path_buf();
         registries_dir.pop();
```

Comment:
Sorry, my suggestion was off; this `pop` needs to be removed too; `.parent()` does the same thing.